Week1/homework/app.js

```diff
@@ -1,11 +1,165 @@
+/* eslint-disable camelcase */
+
 'use strict';
 
 {
+  // #region 1.1_&_1.2
   const bookTitles = [
-    // Replace with your own book titles
-    'harry_potter_chamber_secrets',
+    'The 7 Habits of Highly Effective People',
+    'Animal Farm',
+    'Its Your Ship',
+    'How to Win Friends and Influence People',
+    'Good to Great',
+    'Sapiens',
+    'Little Prince',
+    'The Alchemist',
+    'The Art of War',
+    'Elfabe',
   ];
 
-  // Replace with your own code
-  console.log(bookTitles);
+  /*
+  bookTitles.forEach(item => {
+    console.log(item);
+  });
+  */
+  // #endregion 1.1_&_1.2
+
+  // #region 1.3
+  function idNameConvention(name) {
+    const reg = /\w+/gi;
+    const myString = name.replace(reg, function(s) {
+      return s.toLowerCase();
+    });
+    return myString.replace(/ /gi, '_');
+  }
+
+  function generate_List() {
+    const ul = document.createElement('ul');
+    bookTitles.forEach(item => {
+      const li = document.createElement('li');
+      li.innerHTML = item;
+      li.id = idNameConvention(item);
+      ul.appendChild(li);
+    });
+
+    document.body.appendChild(ul);
+  }
+
+  generate_List();
+  // #endregion 1.3
+
+  // #region 1.4
+  const bookDetails = ['Title', 'Language', 'Author'];
+
+  const Book = function(details, info) {
+    for (let i = 0; i < details.length; i++) {
+      const propName = details[i].toLowerCase();
+      this[propName] = info[i];
+    }
+  };
+
+  function addBook(bookList, book) {
+    bookList[idNameConvention(book.title) + '-2'] = book;
+  }
+
+  function createLibrary() {
+    const library = {};
+    const book_Infos = [
+      [bookTitles[0], 'English', 'Stephan Covey'],
+      [bookTitles[1], 'English', 'George Orwell'],
+      [bookTitles[2], 'English', 'D. Michael Abrashoff'],
+      [bookTitles[3], 'English', 'Dale Carnegie'],
+      [bookTitles[4], 'English', 'Jim Collins'],
+      [bookTitles[5], 'English', 'Yuval Noah Harari'],
+      [bookTitles[6], 'English', 'Antoine de Saint-Exupéry'],
+      [bookTitles[7], 'English', 'Paulo Coelho'],
+      [bookTitles[8], 'English', 'Sun Tzu'],
+      [bookTitles[9], 'Turkish', 'Mehmet Ali Bulut'],
+    ];
+
+    for (let index = 0; index < book_Infos.length; index++) {
+      const b = new Book(bookDetails, book_Infos[index]);
+      addBook(library, b);
+    }
+    return library;
```

Comment:
I now see that this method creates an Object in the shape that was asked for. Although it is a creative way of doing it, it is also a lot more complex for no benefit.

Reply:
Right. I just confused and couldn't think easier.